Tests/SonarScanner.MSBuild.Tasks.IntegrationTest/TargetsTests/TargetsTestsContext.cs

```diff
@@ -0,0 +1,97 @@
+﻿/*
+ * SonarScanner for .NET
+ * Copyright (C) 2016-2025 SonarSource SA
+ * mailto: info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Runtime.InteropServices;
+
+namespace SonarScanner.MSBuild.Tasks.IntegrationTest.TargetsTests;
+
+public class TargetsTestsContext
+{
+    private readonly string language;
+
+    public TestContext TestContext { get; }
+    public string ProjectFolder { get; }
+    public string ConfigFolder { get; }
+    public string InputFolder { get; }
+    public string OutputFolder { get; }
+
+    public TargetsTestsContext(TestContext testContext, string language = "C#", string inputFolder = "Inputs")
+    {
+        this.language = language;
+        TestContext = testContext;
+        ProjectFolder = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext);
+        ConfigFolder = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext, "Config");
+        InputFolder = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext, inputFolder);
+        OutputFolder = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext, "Outputs");
+    }
+
+    public string CreateInputFile(string fileName)
+    {
+        var filePath = Path.Combine(InputFolder, fileName);
+        File.WriteAllText(filePath, null);
+        return filePath;
+    }
+
+    public string CreateProjectFile(string testSpecificProjectXml, AnalysisConfig config = null, bool emptySqProperties = false)
```

Comment:
```suggestion
    public string CreateProjectFile(string testSpecificProjectXml, AnalysisConfig config = null, bool defaultSqProperties = true)
```
and flip the logic in the body. This should make it easier to understand.